File: main/src/main/java/lavaplayer/source/youtube/YoutubePlaylistLoader.java
package lavaplayer.source.youtube;

import lavaplayer.tools.io.HttpInterface;
import lavaplayer.track.AudioTrack;
import lavaplayer.track.AudioTrackInfo;
import lavaplayer.track.AudioTrackCollection;

import java.util.function.Function;

public interface YoutubePlaylistLoader {
    void setPlaylistPageCount(int playlistPageCount);

    AudioTrackCollection load(HttpInterface httpInterface,
                              String playlistId,
                              String selectedVideoId,
                              Function<AudioTrackInfo, AudioTrack> trackFactory);
}

File: main/src/main/java/lavaplayer/source/youtube/YoutubePlaylistLoader.kt
package lavaplayer.source.youtube

import lavaplayer.tools.io.HttpInterface
import lavaplayer.track.AudioTrackCollection
import lavaplayer.track.AudioTrackFactory

interface YoutubePlaylistLoader {
    var playlistPageCount: Int

    fun load(
        httpInterface: HttpInterface,
        playlistId: String,
        selectedVideoId: String?,
        trackFactory: AudioTrackFactory
    ): AudioTrackCollection
}
